lib/engine/game/g_18_usa/step/buy_train.rb

```diff
@@ -13,6 +13,9 @@ def should_buy_train?(entity)
 
           def buyable_trains(entity)
             buyable_trains = super
+            # Cannot buy pullmans from the bank/depot? # TODO: Confirm if this is true and if false, remove this line.
+            # https://github.com/tobymao/18xx/issues/7097
+            buyable_trains.reject! { |t| @game.pullman_train?(t) && t.from_depot? }
```

Comment:
I recommend posting the question of bgg

Reply:
The new version of the rules clarifies this case